clients/src/main/java/org/apache/kafka/common/protocol/types/Schema.java

```diff
@@ -83,10 +92,21 @@ public Struct read(ByteBuffer buffer) {
      */
     @Override
     public int sizeOf(Object o) {
+        return sizeOf(o, false);
+    }
+
+    /**
+     * Size of the raw object which excludes default values.
+     */
+    public int sizeOfRaw(Object o) {
+        return sizeOf(o, true);
+    }
+
+    private int sizeOf(Object o, boolean raw) {
         int size = 0;
         Struct r = (Struct) o;
         for (int i = 0; i < fields.length; i++)
-            size += fields[i].type.sizeOf(r.get(fields[i]));
+            size += fields[i].type.sizeOf(raw ? r.getRaw(fields[i]) : r.get(fields[i]));
```

Comment:
Yep, you're right. I was trying to get away with a change that was more limited in scope, but yeah, that won't work.